cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -89,16 +93,17 @@ public void init(HandlerContext context) {
   }
 
   /**
-   * Sends control commands to the client.
+   * Sends control commands to the client. Also receives program status updates from the client.
    */
-  @GET
+  @PUT
```

Comment:
a PUT normally means you're overwriting the resource that is identified by /tethering/controlchannels/{peer}. This is more like a POST, since it is adding to a resource